### PR TITLE
log what ips we got and make defer pickup when no ips are found

### PR DESCRIPTION
**Pull request by paulgmiller**
**Reason for Change**:

We had an incident where it seems like nmagent returned an empty list of ip address.
Improve logging so we know next time

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests

**Notes**:

### CODE REVIEW

File: ipam/azure.go [hunk 1]
```diff
@@ -172,7 +173,9 @@ func (s *azureSource) refresh() error {
 					log.Printf("[ipam] Failed to create address:%v err:%v.", address, err)
 					continue
 				}
+				addesscount++
 			}
+			log.Printf("[ipam] got %d addresses from interface %s, subnet %s", ifName, subnet)
```
Comment (paulgmiller) [inline ipam/azure.go:178]: Could also reject pools with empty address lists in setaddressspace below.

File: ipam/azure.go [hunk 2]
```diff
@@ -172,7 +173,9 @@ func (s *azureSource) refresh() error {
 					log.Printf("[ipam] Failed to create address:%v err:%v.", address, err)
 					continue
 				}
+				addesscount++
```
Comment (weng271190436) [inline ipam/azure.go:176]: misspelled

File: ipam/pool.go [hunk 3]
```diff
@@ -515,7 +515,8 @@ func (ap *addressPool) requestAddress(address string, options map[string]string)
 		}
 
 		if ar == nil {
-			return "", errNoAvailableAddresses
+			err = errNoAvailableAddresses
+			return "", err
```
Comment (thatmattlong) [inline ipam/pool.go:519]: What is the purpose of this change? style?

**Review by thatmattlong — DISMISSED** [review]

File: ipam/pool.go (hunk 3 above)
Comment (paulgmiller) [reply ipam/pool.go:519]: Not a style thing. It's only so the deferd log statement above picks it up. It was confusing looking at the logs and seeing the function returned nil, nil when that wasn't true

File: ipam/azure.go (hunk 1 above)
Comment (weng271190436) [reply ipam/azure.go:178]: maybe log error

**Review by weng271190436 — APPROVED** [review]

File: ipam/azure.go (hunk 1 above)
Comment (paulgmiller) [reply ipam/azure.go:178]: Punting on this since nobody asked for it.

**Review by matmerr — APPROVED** [review]
/azp run

**Review by matmerr — APPROVED** [review]
